kernel/src/transaction/mod.rs

```diff
@@ -1,23 +1,36 @@
 use std::collections::HashSet;
 use std::iter;
+use std::ops::Deref;
 use std::sync::{Arc, LazyLock};
 use std::time::{SystemTime, UNIX_EPOCH};
 
-use crate::actions::{get_log_add_schema, get_log_commit_info_schema, get_log_txn_schema};
-use crate::actions::{CommitInfo, SetTransaction};
+use url::Url;
+
+use crate::actions::domain_metadata::domain_metadata_configuration;
+use crate::actions::{
+    as_log_add_schema, get_log_commit_info_schema, get_log_domain_metadata_schema,
+    get_log_txn_schema,
+};
+use crate::actions::{CommitInfo, DomainMetadata, SetTransaction};
 use crate::error::Error;
-use crate::expressions::UnaryExpressionOp;
+use crate::expressions::{ArrayData, Transform, UnaryExpressionOp::ToJson};
 use crate::path::ParsedLogPath;
-use crate::schema::{MapType, SchemaRef, StructField, StructType};
+use crate::row_tracking::{
+    RowTrackingDomainMetadata, RowTrackingVisitor, ROW_TRACKING_DOMAIN_NAME,
+};
+use crate::schema::{ArrayType, MapType, SchemaRef, StructField, StructType};
 use crate::snapshot::Snapshot;
 use crate::{
-    DataType, DeltaResult, Engine, EngineData, Expression, ExpressionRef, IntoEngineData, Version,
+    DataType, DeltaResult, Engine, EngineData, Expression, ExpressionRef, IntoEngineData,
+    RowVisitor, Version,
 };
 
-use url::Url;
+/// Type alias for an iterator of [`EngineData`] results.
+type EngineDataIterator<'a> =
+    Box<dyn Iterator<Item = DeltaResult<Box<dyn EngineData>>> + Send + 'a>;
 
-/// The static instance referenced by [`add_files_schema`].
-pub(crate) static ADD_FILES_SCHEMA: LazyLock<SchemaRef> = LazyLock::new(|| {
+/// The minimal (i.e., mandatory) fields in an add action.
+pub(crate) static BASIC_ADD_FILE_SCHEMA: LazyLock<SchemaRef> = LazyLock::new(|| {
```

Comment:
nit: might be better then:
```suggestion
pub(crate) static MANDATORY_ADD_FILE_SCHEMA: LazyLock<SchemaRef> = LazyLock::new(|| {
```